include/windows/winsock2.h

```diff
@@ -0,0 +1,119 @@
+/**
+ * This file has no copyright assigned and is placed in the Public Domain.
+ * This file is part of the mingw-w64 runtime package.
+ * No warranty is given; refer to the file DISCLAIMER.PD within this package.
+ */
+
+#ifndef _WINSOCK2API1_
+#define _WINSOCK2API1_
+
+#include <_mingw_unicode.h>
+
+#ifndef _WINSOCKAPI_
+#define _WINSOCKAPI_
+#else
+#warning Please include winsock2.h before windows.h
+#endif
+
+#ifndef _INC_WINDOWS
+#include <windows.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FROM_PROTOCOL_INFO (-1) 
+
+#define MAX_PROTOCOL_CHAIN 7
+
+#define BASE_PROTOCOL 1
+#define LAYERED_PROTOCOL 0
+
+#define INVALID_SOCKET	(SOCKET)(~0)
+#define SOCKET_ERROR	(-1) 
+
+typedef UINT_PTR	SOCKET; 
+typedef unsigned int GROUP; 
+  
+typedef struct _WSAPROTOCOLCHAIN {
+	int ChainLen;
+
+	DWORD ChainEntries[MAX_PROTOCOL_CHAIN];
+} WSAPROTOCOLCHAIN,*LPWSAPROTOCOLCHAIN; 
+  
+#define WSAPROTOCOL_LEN 255
+
+  typedef struct _WSAPROTOCOL_INFOA {
+    DWORD dwServiceFlags1;
+    DWORD dwServiceFlags2;
+    DWORD dwServiceFlags3;
+    DWORD dwServiceFlags4;
+    DWORD dwProviderFlags;
+    GUID ProviderId;
+    DWORD dwCatalogEntryId;
+    WSAPROTOCOLCHAIN ProtocolChain;
+    int iVersion;
+    int iAddressFamily;
+    int iMaxSockAddr;
+    int iMinSockAddr;
+    int iSocketType;
+    int iProtocol;
+    int iProtocolMaxOffset;
+    int iNetworkByteOrder;
+    int iSecurityScheme;
+    DWORD dwMessageSize;
+    DWORD dwProviderReserved;
+    CHAR szProtocol[WSAPROTOCOL_LEN+1];
+  } WSAPROTOCOL_INFOA,*LPWSAPROTOCOL_INFOA;
+
```

Comment:
Ze zijn bijgeleverd, maar zitten in w32api/winsock2.h. Daarin zit echter ook de windows versie van de socket functies die incompatible zijn met de POSIX versies